main.go

```diff
@@ -158,3 +166,30 @@ func main() {
 		os.Exit(1)
 	}
 }
+
+func applyPrometheusRBAC(client client.Client, namespace string) error {
```

Comment:
Ideally we resync this stuff periodically. Do we miss any edge cases if we sync RBACs only in the beginning?

Reply:
I don't think we miss any edge cases while syncing these RBACs only in the beginning. If I'm not wrong, having them in the manifests, would also means these RBACS will be applied in the beginning only (by ART) , right ?

Reply:
If some other entity changes the RBAC, the operator would have to be restarted to fix the RBAC. I guess the potential for divergence from desired state is the lowest possible(user intervention required), and the desired state only changes with the image which would trigger a restart anyway.

One downside I can think of is if there is some transient issue with the apiserver, the pod could go into CrashLoopBackoff. ([max 5 minutes](https://github.com/kubernetes/kubernetes/blob/5f920426103085a28069a1ba3ec9b5301c19d075/pkg/kubelet/kubelet.go#L121))

Reply:
Putting it in a reconcile loop would mitigate both of those issues, but would add another informer.

Reply:
yeah I do not like relying on pod crash for resync of RBAC rules. There should be a better way. :(

Reply:
@gnufied  @rohantmp  So possible options here :
- Have a new controller to sycn prometheus rbac.  Seems like an overkill to have it just for Prometheus rbacs.
- Sync prometheus rbac in `DaemonReconciler` and `LocalVolumeDiscoveryReconciler` reconciler.  This leads to repetition. But we can't avoid that since both the discovery and lvs/lv daemonset had different reconcilers.
Thoughts?